decoder.go

```diff
@@ -275,31 +302,36 @@ func (d *Decoder) decode(v reflect.Value, path string, parts []pathPart, values
 				}
 			}
 		} else {
-			// When there's no registered conversion for the custom type, we will check if the type
-			// implements the TextUnmarshaler interface. As the UnmarshalText function should be applied
-			// to the pointer of the type, we convert the value to pointer.
-			if v.CanAddr() {
-				v = v.Addr()
-			}
-
-			if u, ok := v.Interface().(encoding.TextUnmarshaler); ok {
-				if err := u.UnmarshalText([]byte(val)); err != nil {
-					return ConversionError{
-						Key:   path,
-						Type:  t,
-						Index: -1,
-						Err:   err,
-					}
-				}
-
-			} else {
-				return fmt.Errorf("schema: converter not found for %v", t)
-			}
+			return fmt.Errorf("schema: converter not found for %v", t)
 		}
 	}
 	return nil
 }
 
+func isTextUnmarshaler(v reflect.Value) (encoding.TextUnmarshaler, bool) {
+	// As the UnmarshalText function should be applied
+	// to the pointer of the type, we convert the value to pointer.
+	if v.CanAddr() {
+		v = v.Addr()
+	}
+	if u, ok := v.Interface().(encoding.TextUnmarshaler); ok {
+		return u, ok
+	}
+
+	// if v is []T or *[]T create new T
+	t := v.Type()
+	if t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+	if t.Kind() == reflect.Slice {
+		t = t.Elem()
```

Comment:
Not sure I understand reflection well enough in go.

Replacing `t = t.Elem()` above with:

```go
if t = t.Elem(); t.Kind() == reflect.Ptr {
    t = t.Elem()
}
```
I get `panic: reflect.Set: value of type schema.rudeBool is not assignable to type *schema.rudeBool`

and with:

```go
if t = t.Elem(); t.Kind() == reflect.Ptr {
    t = reflect.PtrTo(t.Elem())
}
```
I get nil which fails the test for the slice pointer array